src/content/1.7/basics/keeping_up-to-date/migration.md

```diff
@@ -0,0 +1,678 @@
+
+##### mysqldump
+If you used `mysqldump` for your backups, there is also an option for exctracting your data in CSV files, as described in the official documentation: https://dev.mysql.com/doc/refman/8.0/en/mysqldump-delimited-text.html. However, our tests have shown that method is far from being easy.
+
+Example for exporting the table ps_product:
+```bash
+mysqldump -h127.0.0.1 -P3307 -uroot -p -T/var/lib/mysql-files --fields-enclosed-by=\" --fields-terminated-by=\; prestashop16 ps_product
+```
+
+Several notes about `mysqldump` with CSV files:
+
+* Make sure you set the option `--fields-terminated-by` with the separators you want
+* The header containing column names is missing from the .txt file and have to be manually inserted
+
+Here is an extract of the product table, with demonstration data. Only the data is present and we need to add the column names.
+
+{{< figure src="/images/1.7/upgrade-migration/migration-mysqldump-export-raw-csv.png" title="CSV from mysqldump without header" >}}
+
+Look at the SQL file provided with the TXT file, or run a DESCRIBE of the table.
+
+```sql
+mysql> DESCRIBE prestashop16.ps_product;
+```
+
+You should expect the following result before proceeding to the next table:
+
+{{< figure src="/images/1.7/upgrade-migration/migration-mysqldump-export-with-header-csv.png" title="CSV from mysqldump with header" >}}
+
+* You can rename the generated .txt files with the .csv extension to avoid confusion.
+* If you are connected to a remote MySQL server: switching on CSV export will write on the remote filesystem and not on the client one. That’s why writing on a nonexistent or disallowed folder may trigger this error:
+
+```txt
+mysqldump: Got error: 1290: The MySQL server is running with the --secure-file-priv option so it cannot execute this statement when executing 'SELECT INTO OUTFILE'
+```
+
+If you want to know in which folder you can export the files, run the SQL request:
+
+```sql
+mysql> SHOW VARIABLES LIKE "secure_file_priv";
+```
+
+### Updating the data
+
+#### Get changes between source and destination databases
+Now you have retrieved your data from the production shop, you have to adapt it to the new structure before inserting it on your new database.
+
+An efficient way to get the changes is getting a database structure diff between the 2 running shops. MySQL provides another tool for this called [mysqldiff](https://dev.mysql.com/doc/mysql-utilities/1.6/en/mysqldiff.html).
+
+Note this utility may be not immediately available on your environment, even if you have the mysql client installed. For instance on Ubuntu, an additional package needs to be installed:
+
+```bash
+apt install mysql-utilities
+```
+
+This tool can be run anywhere, as long as you can access both servers & databases. The basic use of this command requires the following parameters:
+
+```txt
+Usage: mysqldiff --server1=user:pass@host:port:socket --server2=user:pass@host:port:socket db3:db4
```

Comment:
Consider adding line breaks using`\` to improve readability and avoid horizontal scrolling